### PR TITLE
Exposed task kill -w to cli

### PR DESCRIPTION
**Pull request by Pbarbosa92**
Referent to issue [#75 on tasks](https://github.com/inductiva/tasks/issues/75).

### CODE REVIEW

File: inductiva/_cli/cmd_tasks/kill.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+"""Kills a tasks by id via CLI."""
+
+import inductiva
+
+
+def kill_task(args):
+    """ Kills a task by id.
+    """
```
Comment (IvanPombo) [inline inductiva/_cli/cmd_tasks/kill.py:8]:
```suggestion
    """Kills a task by id."""
```

File: inductiva/_cli/cmd_tasks/kill.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    inductiva.tasks.Task(args.id).kill(wait_timeout=args.wait)
+
+
+def register(parser):
+    """Register the kill task command."""
+
+    subparser = parser.add_parser("kill", help="kills a task.")
```
Comment (IvanPombo) [inline inductiva/_cli/cmd_tasks/kill.py:15]:
```suggestion
    subparser = parser.add_parser("kill", help="Kill a running task.")
```

File: inductiva/_cli/cmd_tasks/kill.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    subparser.add_argument("id", type=str, help="Id of the task to kill.")
+    subparser.add_argument("-w",
+                           "--wait",
```
Comment (IvanPombo) [inline inductiva/_cli/cmd_tasks/kill.py:18]: This should reflect the actual argument in the function -> `wait_timeout`

File: inductiva/_cli/cmd_tasks/kill.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+                           type=float,
+                           default=None,
+                           help="Wait timer for kill to complete.")
```
Comment (IvanPombo) [inline inductiva/_cli/cmd_tasks/kill.py:21]: Timeout to wait for the kill to complete.

**Review by IvanPombo — COMMENTED** [review]
Micro comments.

**Review by IvanPombo — APPROVED** [review]
LGTM

File: inductiva/_cli/cmd_tasks/kill.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+"""Kills a tasks by id via CLI."""
+
+import inductiva
+
+
+def kill_task(args):
+    """Kills a task by id."""
+    inductiva.tasks.Task(args.id).kill(wait_timeout=args.wait)
```
Comment (IvanPombo) [inline inductiva/_cli/cmd_tasks/kill.py:8]: This needs to be `args.wait_timeout` since we changed the argument below.

**Review by IvanPombo — CHANGES_REQUESTED** [review]
Please address the argument name and to improve the experience of killing with CLI, we should add on the `kill-request-sent` message snippet in `localization/translations` the ability to track the task status via CLI with `inductiva tasks -id {task_id}`. As is one can only do it via Python, which doesn't make so much sense when we kill via CLI:

File: inductiva/_cli/cmd_tasks/kill.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+def kill_task(args):
+    """Kills a task by id."""
+    inductiva.tasks.Task(args.id).kill(wait_timeout=args.wait_timeout)
+
+
+def register(parser):
+    """Register the kill task command."""
+
+    subparser = parser.add_parser("kill", help="Kill a running task.")
+    subparser.add_argument("id", type=str, help="Id of the task to kill.")
```
Comment (sergiomsantos) [inline inductiva/_cli/cmd_tasks/kill.py:15]: We should be able to kill multiple tasks at the same time. I suggest setting nargs='+':

```suggestion
    subparser.add_argument("id", type=str, help="ID(s) of the task to kill.", nargs='+')
```

Also, we're missing confirmation request. Something like:

```
Do you confirm you want to kill %d tasks (y/[N])?
```

**IMPORTANT:** check in the CLI document what is the name of the items (`task_id` vs `id`) to make sure we're coherent across all commands.

File: inductiva/_cli/cmd_tasks/kill.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    subparser.add_argument("-w",
+                           "--wait-timeout",
+                           type=float,
+                           default=None,
+                           help="Timeout to wait for the kill to complete.")
+
```
Comment (sergiomsantos) [inline inductiva/_cli/cmd_tasks/kill.py:21]: The user should be able to confirm he want to kill all tasks:

```suggestion
subparser.add_argument("-y", "--yes", action='store_true', ...)
```

**Review by sergiomsantos — COMMENTED** [review]
missing:
- action confirmation;
- possibility of killing multiple tasks at once.

File: inductiva/_cli/cmd_tasks/kill.py (hunk 7 above)
Comment (IvanPombo) [reply inductiva/_cli/cmd_tasks/kill.py:21]: Does the confirmation work for the multiple tasks or should be for each individual one?

File: inductiva/_cli/cmd_tasks/kill.py (hunk 6 above)
Comment (Pbarbosa92) [reply inductiva/_cli/cmd_tasks/kill.py]: We call it id.

File: inductiva/_cli/cmd_tasks/kill.py (hunk 7 above)
Comment (Pbarbosa92) [reply inductiva/_cli/cmd_tasks/kill.py:21]: I've made it one confirmation for all.

File: inductiva/_cli/cmd_tasks/kill.py (hunk 5 above)
Comment (Pbarbosa92) [reply inductiva/_cli/cmd_tasks/kill.py]: Changed. I algo added 
"> Or via the CLI by calling inductiva tasks -id <task id>."
to the translations file.

File: inductiva/_cli/cmd_tasks/kill.py [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        inductiva.tasks.Task(task_id).kill(wait_timeout=args.wait_timeout)
+
+
+def register(parser):
+    """Register the kill task command."""
+
+    subparser = parser.add_parser("kill", help="Kill a running task.")
+    subparser.add_argument("id",
+                           type=str,
+                           help="ID(s) of the task to kill.",
```
Comment (sergiomsantos) [inline inductiva/_cli/cmd_tasks/kill.py:29]:
```suggestion
                           help="ID(s) of the task(s) to kill.",
```

File: inductiva/_cli/cmd_tasks/kill.py [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+    for task_id in args.id:
+        print("Killing task ", task_id)
+        inductiva.tasks.Task(task_id).kill(wait_timeout=args.wait_timeout)
+
+
+def register(parser):
+    """Register the kill task command."""
+
+    subparser = parser.add_parser("kill", help="Kill a running task.")
```
Comment (sergiomsantos) [inline inductiva/_cli/cmd_tasks/kill.py:26]:
```suggestion
    subparser = parser.add_parser("kill", help="Kill running tasks.")
```

File: inductiva/localization/translations/en.json [hunk 10]
```diff
@@ -9,6 +9,7 @@
         "The following error occurred:\n  {0}"
     ],
     "kill-request-sent":["Successfully sent kill request for task {0}.",
-        "> Consider tracking the task status by calling task.get_status()."
+        "> Consider tracking the task status by calling task.get_status().",
+        "> Or via the CLI by calling inductiva tasks -id <task id>."
```
Comment (sergiomsantos) [inline inductiva/localization/translations/en.json:13]: A sentence starting with "Or"?
```suggestion
        "> Consider tracking the task status by calling task.get_status()",
        "> or via the CLI using 'inductiva tasks -id {0}'."
```

File: inductiva/_cli/cmd_tasks/kill.py [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+    subparser.add_argument("id",
+                           type=str,
+                           help="ID(s) of the task to kill.",
+                           nargs="+")
+    subparser.add_argument("-w",
+                           "--wait-timeout",
+                           type=float,
+                           default=None,
+                           help="Timeout to wait for the kill to complete.")
```
Comment (sergiomsantos) [inline inductiva/_cli/cmd_tasks/kill.py:35]: what are the options? See what we have in the `task.kill` method docstring. It should be mapped here.

File: inductiva/_cli/cmd_tasks/kill.py [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+                       f"kill {len(args.id)} tasks (y/[N])?")
+        confirm = prompt.lower() in ["y", "ye", "yes"]
+        if confirm:
+            print("Killing tasks.")
+        else:
+            print("Aborting the killing of tasks.")
+            return
+
+    for task_id in args.id:
+        print("Killing task ", task_id)
```
Comment (sergiomsantos) [inline inductiva/_cli/cmd_tasks/kill.py:19]: we already have this in the kill command.
```suggestion
```

File: inductiva/_cli/cmd_tasks/kill.py [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+    """Kills a task by id."""
+    if not args.yes:
+        prompt = input("Do you confirm you want to "
+                       f"kill {len(args.id)} tasks (y/[N])?")
+        confirm = prompt.lower() in ["y", "ye", "yes"]
+        if confirm:
+            print("Killing tasks.")
+        else:
+            print("Aborting the killing of tasks.")
+            return
```
Comment (sergiomsantos) [inline inductiva/_cli/cmd_tasks/kill.py:16]: Remove most of it. The user already has all feedback from his immediate actions.
```suggestion
        if not confirm:
            print("Aborted.")
```

**Review by sergiomsantos — APPROVED** [review]

**Comment by Pbarbosa92** [comment]
**QA**
Test the cli task kill command.
```
usage: inductiva tasks kill [-h] [-w WAIT_TIMEOUT] [-y] id [id ...]

positional arguments:
  id                    ID(s) of the task(s) to kill.

options:
  -h, --help            show this help message and exit
  -w WAIT_TIMEOUT, --wait-timeout WAIT_TIMEOUT
                        Number of seconds to wait for the kill command. If not provided, the system sends the request without waiting a response.
  -y, --yes             Skip kill confirmation.

```
Don't forget to kill multiple tasks at once like this inductiva tasks kill id1 id2 id3 -y -w 1

@IvanPombo and @AugustoPeres

**Comment by AugustoPeres** [comment]
QA ✅ 
Screenshot 2024-02-01 at 17 50 16
Screenshot 2024-02-01 at 17 51 52

**Pull request merged**